InfiniteLoopers-App/src/main/java/com/isa/account/RoleAssinger.java

```diff
@@ -0,0 +1,44 @@
+package com.isa.account;
+
+import java.util.List;
+
+public class RoleAssinger {
+    private UserInputReader userInputReader = new UserInputReader();
+
+    public void assingRoles(List<User> userList, AdminUser admin) {
+        System.out.println("User: ");
+        for (User user : userList) {
+            System.out.println(user.getAgeNewUser());
+        }
+
+        String selectUserLogin = userInputReader.readNonEmptyString("Wprowadz login do ktorego chcesz przepisac role: ");
+
+        User selectedUser = null;
+        for (User user : userList) {
+            if (user.getLoginNewUser().equals(selectUserLogin)) {
+                selectedUser = user;
+                break;
+            }
+        }
+
+        if (selectedUser != null) {
+            System.out.println("Dodana rola: ");
+            for (UserRole role : UserRole.values()) {
+                System.out.println(role);
+            }
+
+            String selectedRoleStr = userInputReader.readNonEmptyString("Wprowadz role user: ");
+
+            try {
+                UserRole selectedRole = UserRole.valueOf(selectedRoleStr);
+                admin.addRoleToUser(selectedUser, selectedRole);
+                System.out.println("Rola dodana!");
+            } catch (IllegalArgumentException error) {
+                System.out.println("Blad w prowadzeniu roli.");
+            }
+        } else {
+            System.out.println("User nie zostal znaleziony");
+        }
```

Comment:
to wszystko bym zamienił na opitonala, tę część z `ifPresent`, czyli od sout do catch można wydzielić do osobnej funkcji, a bez tego wyglądałoby to mniej więcej tak:
```
userList.stream()
    .filter(user -> user.getLoginNewUser().equals(selectUserLogin))
    .findFirst()
    .ifPresentOrElse(user -> {
        System.out.println("Dodana rola: ");
        for (UserRole role : UserRole.values()) {
            System.out.println(role);
        }

        String selectedRoleStr = userInputReader.readNonEmptyString("Wprowadz role user: ");

        try {
            UserRole selectedRole = UserRole.valueOf(selectedRoleStr);
            admin.addRoleToUser(user, selectedRole);
            System.out.println("Rola dodana!");
        } catch (IllegalArgumentException error) {
            System.out.println("Blad w prowadzeniu roli.");
        }
    },
    () -> System.out.println("User nie zostal znaleziony"));
```

Reply:
Tego nie rozumiem, poproszę wytłumaczyć na zajęciach projektowych